manifests/user.pp

```diff
@@ -205,139 +203,131 @@
 ) {
 
   assert_type(Accounts::User::Name, $name)
-  if $home {
-    $_home = $home
-  } elsif $name == 'root' {
-    $_home = $::osfamily ? {
-      'Solaris' => '/',
-      default   => '/root',
-    }
-  } else {
-    $_home = $::osfamily ? {
-      'Solaris' => "/export/home/${name}",
-      default   => "/home/${name}",
-    }
+
+  $_home = $home ? {
+    undef => $name ? {
+      'root' => $facts['os']['family'] ? {
+        'Solaris' => '/',
+        default   => '/root',
+      },
+      default => $facts['os']['family'] ? {
+        'Solaris' => "/export/home/${name}",
+        default   => "/home/${name}",
+      },
+    },
+    default => $home,
   }
 
-  if $locked {
-    case $::operatingsystem {
-      'debian', 'ubuntu' : {
-        $_shell = '/usr/sbin/nologin'
-      }
-      'solaris' : {
-        $_shell = '/usr/bin/false'
+  if $ensure == 'absent' {
+    user { $name:
+      ensure     => 'absent',
+      forcelocal => $forcelocal,
+      home       => $_home,
+      managehome => $purge_user_home,
+    }
+    User[$name] -> Group <| |>
+    if $purge_user_home {
+      accounts::home_dir { $_home:
+        ensure => 'absent',
+        group  => $group,
+        user   => $name,
+
       }
-      default : {
-        $_shell = '/sbin/nologin'
+    }
+    if $create_group {
+      # Only remove the group if it is the same as user name as it may be shared
+      if $name == $group and !defined(Group[$group]) {
```

Comment:
Should this condition not be `$name == $group and defined(Group[$group])` in order to mark the group which should already be defined as absent?

Reply:
It would be an error, in Puppet, to define a resource that has already been defined.

Perhaps it would be better to use [`ensure_resource`](https://github.com/puppetlabs/puppetlabs-stdlib#ensure_resource) instead of duplicating its functionality here.